src/hourly-apportioned-emissions/hour-unit-data.repository.ts

```diff
@@ -110,39 +114,50 @@ export class HourUnitDataRepository extends Repository<HourUnitData> {
 
       string += ')';
 
-      results.andWhere(string);
+      query.andWhere(string);
     }
 
     if (unitFuelType) {
       let string = '(';
 
       for (let i = 0; i < unitFuelType.length; i++) {
-
         const regex = `'((^${unitFuelType[i].toUpperCase()
         }$)|([,][ ]*${unitFuelType[i].toUpperCase()
         }$)|([,][ ]*${unitFuelType[i].toUpperCase()
         }[,])|(^${unitFuelType[i].toUpperCase()
         }[,])|(^${unitFuelType[i].toUpperCase()
         } [(])|([,][ ]*${unitFuelType[i].toUpperCase()} [(]))'`;
```

Comment:
DRY (Don't Repeat Yourself) design principal should be followed here. Since these lines are repeated a 2nd time within the same method and will probably be needed in other endpoints we need to look at extracting this out into a utility function that takes an array and returns the regex.